### DISCUSSION TITLE
behavior of hx-swap-oob if the target element does not exist

### DISCUSSION THREAD

**Discussion by untitaker** [bigskysoftware/htmx#3009] [Q&A]
consider the following demo:

**index.html**

```
<script src="https://unpkg.com/htmx.org@2.0.3"></script>

<h2 id=heading hx-swap-oob="true">main</h2>

<div style='border: red' hx-boost="true">
    <main>
        <a href="other.html">other</a>
    </main>
</div>
```

**other.html**

```
<script src="https://unpkg.com/htmx.org@2.0.3"></script>

<h2 id=heading hx-swap-oob="true">other</h2>

<div style='border: red' hx-boost="true">
    <main>
        <a href="index.html">main</a>

    </main>
</div>
```

I'd expect that clicking the link swaps out the `h2` and link text accordingly, as `hx-boost` is set up to swap out the entire body.

instead clicking the boosted link makes the header disappear. after I remove `hx-swap-oob`, it works fine.

you might wonder why hx-swap-oob is present at all. that's because on another page I'm using hx-swap to instead of hx-boost to enable a more sophisticated replacement of content. I don't want to use conditions on the server to satisfy both usecases because that means the HTML fragment will be loaded and cached twice.

is this kind of thing possible at all? is there some fallback behavior for `hx-swap-oob` that would leave the element alone if a target ID can't be found in the parent page?

thanks in advance

**Comment by scrhartley** [comment]
I think oob swaps are done first:
so index.html has its heading's content replaced, and other.html would no longer have the `hx-swap-oob` element.
Then the main swap would occur, which targets body by default,
and that would be replaced with the remaining html from other.html, which has already had the `hx-swap-oob` element removed.

**Reply by untitaker** [reply]
Thank you, it seems that `hx-target=main` and `hx-select=main` solves this issue. I wish htmx would warn for this situation

**Reply by maemigh** [reply]
I wanted to ask the same thing. I have a base page that loads from a template and that template also loads many fragments. I may want to render the whole page or I may want to refresh several tiny fragments at once (without reloading a larger portion if the page). On first load this works fine, I guess hx-swap-oob is only running for hx requests? 
But if I then load the same page via an hx-get link, the elements with hx-swap-oob=“true” and no matching element on the page completely disappear. 

I can’t specify hx-target from the originating link because there are multiple targets.

I am thinking it would be nice if there was an option for hx-oob-swap that did enable fallback behavior if the element doesn’t exist which would simply render the element in place as if hx-oob-swap wasn’t specified.
